js/dashboard.js

```diff
@@ -19,55 +120,184 @@
 	 */
 	function updateDashboard(refresh) {
 		// Disable refresh button.
-		if (refreshBtn) {
-			refreshBtn.disabled = true;
+		if (controls.refreshBtn) {
+			controls.refreshBtn.disabled = true;
 		}
 
 		// Load data from API.
-		wp.apiFetch({
-			path: '/statify/v1/stats' + (refresh ? '?refresh=1' : ''),
-		})
+		fetchData('stats', refresh ? 'refresh=1' : null)
 			.then((data) => {
 				const labels = Object.keys(data.visits);
 				const values = Object.values(data.visits);
 
-				render(chartElem, labels, values);
+				render(charts.dashboard, labels, values, false);
 
 				// Render top lists.
-				if (referrerTable) {
-					renderTopList(referrerTable, data.referrer);
+				if (tables.referrer) {
+					renderTopList(tables.referrer, data.referrer);
 				}
-				if (targetTable) {
-					renderTopList(targetTable, data.target);
+				if (tables.target) {
+					renderTopList(tables.target, data.target);
 				}
 
-				if (totalsTable) {
-					renderTotals(totalsTable, data.totals);
-				}
-
-				// Re-enable refresh button.
-				if (refreshBtn) {
-					refreshBtn.disabled = false;
+				if (tables.totals) {
+					renderTotals(tables.totals, data.totals);
 				}
 			})
 			.catch(() => {
 				// Failed to load.
-				chartElem.innerHTML =
+				charts.dashboard.innerHTML =
 					'<p>' +
 					wp.i18n.__('Error loading data.', 'statify') +
 					'</p>';
```

Comment:
Should be possible to write it shorter like that: 

```suggestion
				charts.dashboard.innerHTML = `<p>${wp.i18n.__('Error loading data.', 'statify')}</p>`;
```

Reply:
Updated in all 4 places (3 new + 1 pre-existing). Same for some `'<td>' + ... + '</td>'` blocks.